x/wasm/keeper/handler_plugin.go

```diff
@@ -86,16 +86,21 @@ func (h SDKMessageHandler) handleSdkMessage(ctx sdk.Context, contractAddr sdk.Ad
 		}
 	}
 
-	// find the handler and execute it
-	handler := h.router.Route(ctx, msg.Route())
-	if handler == nil {
-		return nil, sdkerrors.Wrap(sdkerrors.ErrUnknownRequest, msg.Route())
-	}
-	res, err := handler(ctx, msg)
-	if err != nil {
-		return nil, err
+	if legacyMsg, ok := msg.(legacytx.LegacyMsg); ok {
```

Comment:
Does This `legacytx` need to remove or fix?

Reply:
legacytx is used in many places. 
There is a possibility that it will be updated in cosmos-sdk in the future.